sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/util/RLimit.java

```diff
@@ -0,0 +1,87 @@
+package com.mesosphere.sdk.specification.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.mesos.Protos;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Representation of an individual rlimit, consisting of a name and optional soft/hard limits.
+ *
+ * A valid instance of this class has either both limits set or neither, with the further constraint that the soft limit
+ * must be less than or equal to the hard limit.
+ */
+public class RLimit {
+    private static final Set<String> RLIMITS = new HashSet<>();
+
+    static {
+        for (Protos.RLimitInfo.RLimit.Type rLimitType : Protos.RLimitInfo.RLimit.Type.values()) {
+            RLIMITS.add(rLimitType.toString());
+        }
+    }
+
+    private final String name;
+    private final Long soft;
+    private final Long hard;
+
+    public RLimit(
+            @JsonProperty("name") String name,
+            @JsonProperty("soft") Long soft,
+            @JsonProperty("hard") Long hard) throws InvalidRLimitException {
+        this.name = name;
+        this.soft = soft;
+        this.hard = hard;
+
+        validate();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Optional<Long> getSoft() {
+        return Optional.ofNullable(soft);
+    }
+
+    public Optional<Long> getHard() {
+        return Optional.ofNullable(hard);
+    }
+
+    private void validate() throws InvalidRLimitException {
+        if (!RLIMITS.contains(name)) {
+            throw new InvalidRLimitException(name + " is not a valid rlimit of " + RLIMITS.toString());
```

Comment:
Assuming the above `RLMT_` => `RLIMIT_` conversion in expected names is performed, this error could also directly cite the docs with e.g. `See man setrlimit(2)`.